plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -0,0 +1,703 @@
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid fwct data");
+		return FALSE;
+	}
+
+	/* send fwct data */
+	g_debug ("fw write request with meta:%u fwct:%u", (guint32)custom_meta_size, (guint32)fwct_size);
+	if (!fu_ccgx_dmc_device_send_fwct (self,fwct_data, fwct_size, error))
+		return FALSE;
+
+
+	/* init fw written */
+	self->fw_data_written = 0;
+	/* get total fw size */
+	self->fw_data_size = fu_ccgx_dmc_firmware_get_fw_data_size (FU_CCGX_DMC_FIRMWARE(firmware));
+	g_debug ("updating fw...");
+	while (1) {
+		fu_device_set_status (device, FWUPD_STATUS_DEVICE_BUSY);
+		/* get interrupt request */
+		if (!fu_ccgx_dmc_device_read_intr_req (self, &dmc_int_rqt, error))
+			return FALSE;
+		/* fw upgrade request */
+		if (dmc_int_rqt.opcode == DMC_INT_OPCODE_FW_UPGRADE_RQT) {
+			img_index = dmc_int_rqt.data[0];
+			g_debug ("write image #%d",img_index);
+			if (img_index >= image_records->len) {
+				g_set_error (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "invalid image index %d, expected less than %u",
+					     img_index, image_records->len);
+				return FALSE;
+			}
+			fu_device_set_status (device, FWUPD_STATUS_DEVICE_WRITE);
+			img_rcd = g_ptr_array_index (image_records, img_index);
+			/* write image */
+			if (!fu_ccgx_hpi_write_firmware_image (device, img_rcd, error))
+				return FALSE;
+		} else
+			break;
+	}
+	if (dmc_int_rqt.opcode != DMC_INT_OPCODE_FW_UPGRADE_STATUS) {
+		if (dmc_int_rqt.opcode == DMC_INT_OPCODE_FWCT_ANALYSIS_STATUS) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "fwct analysis failed with status = %d",
+				     dmc_int_rqt.data[0]);
+		} else {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "invalid dmc intr req opcode = %d with status = %d",
+				     dmc_int_rqt.opcode, dmc_int_rqt.data[0]);
+	     }
+	     return FALSE;
+	}
+
+	if (dmc_int_rqt.data[0] == DMC_DEVICE_STATUS_UPDATE_PHASE_1_COMPLETE)
+		self->update_model = DMC_UPDATE_MODEL_DOWNLOAD_TIRGGER;
+	else if (dmc_int_rqt.data[0] == DMC_DEVICE_STATUS_FW_DOWNLOADED_UPDATE_PEND)
+		self->update_model = DMC_UPDATE_MODEL_PENDING_RESET;
+	else {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid  status code = %u",
+			     dmc_int_rqt.data[0]);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static FuFirmware *
+fu_ccgx_dmc_device_prepare_firmware (FuDevice *device,
+				     GBytes *fw,
+				     FwupdInstallFlags flags,
+				     GError **error)
+{
+	g_autoptr(FuFirmware) firmware = fu_ccgx_dmc_firmware_new ();
+	FuCcgxDmcDevice *self = FU_CCGX_DMC_DEVICE (device);
+	GBytes* custom_meta_record = NULL;
+	gboolean custom_meta_exist = FALSE;
+
+	/* parse all images */
+	if (!fu_firmware_parse (firmware, fw, flags, error))
+		return NULL;
+
+	/* get custom meta record */
+	custom_meta_record = fu_ccgx_dmc_firmware_get_custom_meta_record (FU_CCGX_DMC_FIRMWARE(firmware));
```

Comment:
space before bracket pls

Reply:
Fixed it on today commit (ccgx: fix code alignment, modify code for review)